### DISCUSSION TITLE
Add "Enable DNS Rewrites" in client settings

### DISCUSSION THREAD

**Discussion by tophrr** [AdguardTeam/AdGuardHome#4093] [Q&A]
After updating to v0.107, I found out that DNS rewrites no longer works if I configure and disable "Block domains using filters and hosts files" in the client setting. Perhaps there would be an option to specifically reenable DNS rewrites on each client? It would be useful for me when I want a specific client to have no adblocking but still have DNS rewrites to access my local services.

**Comment by ainar-g** [comment]
Hello.  The DNS rewrite functionality will largely be deprecated in favour of [`$dnsrewrite` rules][1].  And you can already enable and disable them per client.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

**Reply by igornet1** [reply]
Hello.  Rules `||example123.com^$dnsrewrite=NOERROR;A;1.2.3.4,important,ctag=user_regular`   or  `||example123.com^$dnsrewrite=NOERROR;A;1.2.3.4,important,client=192.168.2.0/24`  also does not work  (The user does not have a checkbox to use global settings since it should not apply filters - blacklists, but user needs global rewrites. User is tagged  user_egular.)  (checked in version 0.108.0-a.53+f9aa5ae8 on the Windows64)  Thank you for your reply and for your work on this project.

**Reply by ainar-g** [reply]
@igornet1, the way to currently achieve that is to allow all requests for the client using an exception rule.  For example:

```none
@@*$client="My Client"
||example.com^$dnsrewrite=NOERROR;A;1.2.3.4,client="My Client"
```

Although I agree that this is a rather roundabout way of doing that.

@ameshkov, what do you think?  Perhaps we should actually separate the DNS rewrite functionality into a separate setting?
